src/tls/server.c

```diff
@@ -680,8 +690,8 @@ server_poll_event (int timeout)
     }
   else if (ret > 0)
     {
-      if (ev.data.ptr == &server)
-        handle_accept ();
+      if (ev.data.ptr >= (void *) server.listen_fds && ev.data.ptr < (void *) &server.listen_fds[MAX_LISTEN_FDS])
```

Comment:
Is the cast necessary?

Reply:
Yes, without it:

    src/tls/server.c:699:23: error: comparison of distinct pointer types lacks a cast [-Werror]

which seems fair. Comparing an int* to a void* (i. e. effectively a byte pointer) should be safe for the array start, but not for any of its subsequent members.